packages/xo-web/src/xo-app/host/tab-advanced.js

```diff
@@ -137,13 +141,52 @@ export default class extends Component {
 
   _setRemoteSyslogHost = value => setRemoteSyslogHost(this.props.host, value)
 
+  _isXCPngHost = () => this.props.host.productBrand === 'XCP-ng'
+
+  _accessAdvancedLiveTelemetry = () =>
+    window.open(
+      `/netdata/${this.props.host.hostname}`
+    )
+
   render() {
     const { host, pcis, pgpus } = this.props
-    const { isHtEnabled } = this.state
+    const { isHtEnabled, isNetDataInstalledOnHost } = this.state
+    const EnableTelemetryButton = () => (
```

Comment:
> You don't need a wrapper component:
> 
> ```js
> const enableTelemetryButton = <TabButton ... />
> ```

HOC components need to be wrapped ! am I missing something ?

Reply:
Yes but there, you only need the instance, not a new component:
```jsx
{isNetDataInstalledOnHost
  ? accessTelemetryButton
  : _isXCPngHost
  ? enableTelemetryButton
  : <Tooltip content={_('xcpOnlyFeature')}>
    <span>
      {enableTelemetryButton}
    </span>
  </Tooltip>
}
```

The only thing though, is that you only need to instantiate one of them on each render. So something like this might even be slightly better:
```js
const telemetryButton = isNetDataInstalledOnHost
  ? <TabButton
    btnStyle='success'
    handler={this._accessAdvancedLiveTelemetry}
    handlerParam={host}
    icon='telemetry'
    labelId='accessAdvancedLiveTelemetry'
  />
  : <TabButton
    btnStyle='success'
    disabled={!_isXCPngHost}
    handler={enableAdvancedLiveTelemetry}
    handlerParam={host}
    icon='telemetry'
    labelId='enableAdvancedLiveTelemetry'
  />
```
and
```jsx
{_isXCPngHost ? (
  telemetryButton
) : (
  <Tooltip content={_('xcpOnlyFeature')}>
    <span>
      {telemetryButton}
    </span>
  </Tooltip>
)}
```

Reply:
done :heavy_check_mark: